backend/src/starknet.ts

```diff
@@ -1,24 +1,78 @@
 import { StarknetIdNavigator } from "starknetid.js";
-import { RpcProvider, constants } from "starknet";
+import {RpcProvider, constants, starknetId} from "starknet";
 import * as dotenv from "dotenv";
 
 dotenv.config();
 
 const NODE_URL: string = process.env.NODE_URL || "";
+const CHAIN_ID: string = process.env.NETWORK === 'mainnet'
+    ? constants.StarknetChainId.SN_MAIN
+    : constants.StarknetChainId.SN_SEPOLIA;
 
-export const getStarknetId = async (
-  address: string
-): Promise<string | null> => {
-  const provider = new RpcProvider({
-    nodeUrl: NODE_URL,
-    chainId: constants.StarknetChainId.SN_MAIN
-  });
+const provider = new RpcProvider({
+  nodeUrl: NODE_URL,
+  chainId: CHAIN_ID,
+});
 
-  const starknetIdNavigator = new StarknetIdNavigator(
+const starknetIdNavigator = new StarknetIdNavigator(
     provider,
-    constants.StarknetChainId.SN_MAIN
-  );
+    CHAIN_ID,
+);
+
+export const getVestingEvents = async (address: string) => {
```

Comment:
This is returning events, but what we would like to receive on the FE is the list of current active vests – why not do the 'compaction' of this on the backend?

Reply:
It's returning `compaction` of event data,
so we will display, claimed, available for claim and future claim
that is why I made this return structure list of dicts:
```
amount: amount,
timestamp: timestamp,
is_claimable: true
```